nemo_skills/inference/model/base.py

```
@@ -99,35 +449,22 @@ def __init__(
-        )
-        httpx_limits = httpx.Limits(max_keepalive_connections=2048, max_connections=2048)
-        litellm.client_session = httpx.Client(limits=httpx_limits)
-        litellm.aclient_session = httpx.AsyncClient(limits=httpx_limits)
+    def _resolve_api_key(self, api_key: str | None, api_key_env_var: str | None, base_url: str) -> str | None:
+        """Resolve API key from various sources"""
+        resolved_key = self._get_api_key(api_key, api_key_env_var, base_url)
+        if resolved_key is None:  # self-hosted models don't need the key
+            resolved_key = "EMPTY"
+        return resolved_key
 
     def _get_api_key(self, api_key: str | None, api_key_env_var: str | None, base_url: str) -> str | None:
```

Ruff (0.13.1)

Unused method argument: `base_url`

(ARG002)